### ISSUE TITLE
[Question] Keeping my monitors off after inactivity

### ISSUE THREAD

**Issue by orfeasxyz** [labwc/labwc#2122]
I am currently using [wlopm](https://git.sr.ht/~leon_plickat/wlopm) and [swayidle](https://github.com/swaywm/swayidle) to turn my monitors off after a certain amount of inactivity. However after I turn them off using "wlopm --off \*", they turn back on just a few seconds later. My goal is to keep them off until there is keyboard/mouse input.

**Comment by daniel005007** [comment]
the same behavior happens to me. here are the logs:

`WAYLAND_DEBUG=server labwc -d`
[labwc-log.txt](https://github.com/user-attachments/files/16826792/labwc-log.txt)

`WAYLAND_DEBUG=1 wlopm --off HDMI-A-1`
[wlopm-off-log.txt](https://github.com/user-attachments/files/16826795/wlopm-off-log.txt)

`WAYLAND_DEBUG=1 wlopm --on HDMI-A-1`
[wlopm-on-log.txt](https://github.com/user-attachments/files/16826797/wlopm-on-log.txt)

**Comment by daniel005007** [comment]
> I would imagine the problem has to do with either swayidle or the interaction between it and the idle protocol.

at least here, I ran wlopm without swayidle.

**Comment by droc12345** [comment]
exactly how are you turning off the monitor? some script, or manually?

I turn  mine off manually and I've never had it turn itself back on.
`<keybind key="W-F12">`
        `<action name="Execute" command="/home/don/bin/wlopm --toggle DP-1" />`

**Comment by daniel005007** [comment]
manually, from the CLI. but the error also occurs when executed by swayidle or a keybind.

**Comment by droc12345** [comment]
Which version of labwc? git or one of the releases?

**Comment by daniel005007** [comment]
release: `labwc 0.8.0`

**Comment by droc12345** [comment]
You have two cards (I can see you use card1), 
is this a regular pc or laptop? how is the monitor connected, video card hdmi port, or running off usb?

I can see from the log where it got turned off then some hotplug events started but I don't know what triggers them off.

**Comment by daniel005007** [comment]
is an regular PC. the iGPU is turned off in the BIOS and the monitor is connected to the dGPU's HDMI port.

**Comment by orfeasxyz** [comment]
Since my original post was too generic and I didn't provide much information here is the testing I have done.

----
Setup:

* Left monitor is an AOC 24G1WG4 connected over HDMI
* Right monitor is a DELL P2419H D9V5K43 connected over DP

----

I tried to run `wlopm` manually:

```bash
$ wlopm --off \* # without swayidle
$ swayidle timeout 5 "wlopm --off \*" # with swayidle
```

Both of these resulted in both monitors turning off momentarily, before turning back on **without** any input. That made me rule out the likelihood of `swayidle` being at fault here. 

Then I tried turning off individual monitors like:

```bash
$ wlopm --off DP-2 # turn off my DELL monitor
$ # turns off until I do a `wlopm --on DP-2`
$ wlopm --off HDMI-A-1 # turn off my AOC monitor
$ # turns back on after a few seconds
```

It seems that both my monitors exhibit a different behavior when powering on/off. This isn't actually new to me. As far back as I remember my AOC monitor never actually disconnected when I powered it off on the fly. In contrast, my DELL monitor did. A very similar issue I've found is from this [post](https://answers.microsoft.com/en-us/windows/forum/all/turning-off-a-monitor-acts-like-disconnecting-it/c38febe1-4b3c-49b8-98ad-c0fcd0dbc5d0) in Microsoft's forum. 

I may switch up cables to see whether the ports have any impact on this.

**Comment by daniel005007** [comment]
it looks like it's a labwc bug. in sway wlopm works correctly (not completely, the monitor is turned off and remains off, even when I run `wlopm --on HDMI-A-1`), but the sway output manager itself (`output HDMI-A-1 power toggle`) works correctly.

**Comment by orfeasxyz** [comment]
I changed from HDMI to DP (My DELL on DP-2 and my AOC on DP-1) and everything seems to be working well now. Maybe there is an issue with the HDMI ports (?).

**Comment by johanmalm** [comment]
Does the `wlopm --on/--off` command work with sway?

**Comment by stefonarch** [comment]
Here `swayidle -w timeout 300 "wlopm --off \*" resume "wlopm --on \*" > /dev/null 2>1 &` in autostart works without issues.

**Comment by archlinux-fan** [comment]
> Here `swayidle -w timeout 300 "wlopm --off \*" resume "wlopm --on \*" > /dev/null 2>1 &` in autostart works without issues.

I have the same code in autostart, but the monitor powers back on in a couple of seconds.

**Comment by letkan** [comment]
I prefer _suspend_, I have this in my config and it works as expected:
`swayidle -w timeout 1000 'systemctl suspend' >/dev/null 2>&1 &`
and this for when I watch movies:
`wayland-pipewire-idle-inhibit >/dev/null 2>&1 &`

**Comment by archlinux-fan** [comment]
> My monitor powers on automatically too after using the `$ wlopm --off \*` I have only 1 monitor connected over HDMI. No dedicated graphic card. Just AMD 680m iGPU.
> 
> I've been struggleing for months trying to solve this issue with no luck. For now I am powering off the monitor manually when I leave the computer for a longer period. I have even considered going back to Openbox, because I didn't have that issue there.

I have installed hyprland (wayland) and the issue persists there too. So I am thinking it's not labwc but wayland related. Or HDMI related.

With DP connection the monitor powered off and stayed off but it never powered back on no matter what key I pressed on the bluetooth keyboard or mouse.

**Comment by Arnaudv6** [comment]
Just mentioning I thought I was hit. OP did not mention kanshi, which I use. This fixed it for me:
https://labwc.github.io/labwc-config.5.html#entry_core_autoenableoutputs

**Comment by Consolatis** [comment]
Yes, this is sadly a weird behavior of some monitors / display drivers and its unlikely that labwc can do anything about it without removing support for `wlopm` and completely handling monitor suspend internally. If you are affected by the issue using kanshi + `autoEnableOutputs` as linked above is the best option from the labwc side.
